### PR TITLE
add new SampleQueries

### PR DESCRIPTION
**Pull request by sevap8**
Hi everyone, sorry for the delay, could not configure LINQ to Sql queries.
Please see how time will be.

### CODE REVIEW

File: HW5/Seva-Popov/Task/LinqSamples.cs [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+            //var result =
+            //    from c in dataSource.Customers
+            //    from o in c.Orders
+            //    orderby o.OrderDate.Month
+            //    group o by o.OrderDate.ToString("MMM") into gr
+            //    select new { gr.Key, total = (from x in gr.ToList() select x.Total).Sum(), count = (from x in gr.ToList() select x.Total).Count() };
+
+            var result = dataSource.Customers.SelectMany(c => c.Orders).OrderBy(o => o.OrderDate.Month).GroupBy(o => o.OrderDate.ToString("MMM")).
+                Select(gr => new { gr.Key, total = gr.ToList().Select(x => x.Total).Sum(), count = gr.ToList().Select(x => x.Total).Count() });
```
Comment (MilRoad) [inline HW5/Seva-Popov/Task/LinqSamples.cs]: It's okay with year statistic but for months you should divide it by years count.

File: HW5/Seva-Popov/Task/LinqSamples.cs [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+        [Description("Client statistics by month of the year")]
+
+        public void LinqHW10C()
+        {
+            var result =
+                 from c in dataSource.Customers
+                 from o in c.Orders
+                 orderby o.OrderDate.Year, o.OrderDate.Month
+                 group o by new { year = o.OrderDate.Year, month = o.OrderDate.ToString("MMM") } into gr
+                 select new { gr.Key.year, gr.Key.month, total = gr.ToList().Select(x => x.Total).Sum(), count = gr.ToList().Select(x => x.Total).Count() };
```
Comment (MilRoad) [inline HW5/Seva-Popov/Task/LinqSamples.cs]: I think it's better to calculate average amount of orders, not its total.

File: HW5/Seva-Popov/Task/LinqToSqlSamples.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        [Description("Give a list of all customers whose total turnover(the sum of all orders) exceeds a certain value X. + LINQ to Sql")]
+
+        public void Linq1()
+        {
+            decimal x = 10000;
+            var customerList =
+                from c in db.Customers
+                where c.Orders.Sum(p => p.Order_Details.Sum(l =>
+                l.UnitPrice * (decimal)(1 - l.Discount) * l.Quantity)) > x
+                select c.CompanyName;
```
Comment (MilRoad) [inline HW5/Seva-Popov/Task/LinqToSqlSamples.cs]: Again you need to print orders total sum

File: HW5/Seva-Popov/Task/LinqToSqlSamples.cs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        [Description("Find all customers who have orders exceeding X in total. + LINQ to Sql")]
+
+        public void Linq2()
+        {
+            decimal x = 3000;
+            var customerList =
+                from c in db.Customers
+                where c.Orders.Any(p => p.Order_Details.Sum(l =>
+                l.UnitPrice * (decimal)(1 - l.Discount) * l.Quantity) > x)
+                select c.CompanyName;
```
Comment (MilRoad) [inline HW5/Seva-Popov/Task/LinqToSqlSamples.cs]: Show orders price

File: HW5/Seva-Popov/Task/LinqToSqlSamples.cs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        [Description("Indicate all customers who have a non-digital postal code or a region is not filled out or an operator code is not indicated on the phone (for simplicity we consider this to be equivalent to “no parentheses at the beginning” + LINQ to Sql")]
+
+        public void Linq3()
+        {
+            var customerList =
+                from customer in db.Customers
+                where System.Data.Linq.SqlClient.SqlMethods.Like(customer.PostalCode.Trim(), "%[^0-9]%")
+                || customer.Region == null
+                || customer.Phone[0] != '('
+                select customer.CompanyName;
```
Comment (MilRoad) [inline HW5/Seva-Popov/Task/LinqToSqlSamples.cs]: It would be better to show there more information (region, phone, postal code)

File: HW5/Seva-Popov/Task/LinqSamples.cs [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+        [Description("Give a list of all customers whose total turnover (the sum of all orders) exceeds a certain value X.")]
+
+        public void Linq1()
+        {
+            int[] x = { 10000, 20000, 30000 };
+
+            foreach (var item in x)
+            {
+                var customerList = dataSource.Customers.Where(q => q.Orders.Sum(w => w.Total) > item)
+                    .Select(e => new { e.CustomerID, e.CompanyName});
```
[the remark this answers is not in the document]
Comment (sevap8) [reply HW5/Seva-Popov/Task/LinqSamples.cs]: Add

File: HW5/Seva-Popov/Task/LinqSamples.cs [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+
+        [Category("Homework")]
+        [Title("Homework - Task 5")]
+        [Description("Do the previous task, but issue a list sorted by year, month, customer turnover (from maximum to minimum) and the name of the client.")]
+
+        public void Linq5()
+        {
+            var customerList = dataSource.Customers.Where(q => q.Orders.Count() > 0)
+                .Select(w => new { DateTime = w.Orders.Min(e => e.OrderDate), CashTurnover = w.Orders.Sum(b => b.Total), w.CompanyName })
+                .OrderBy(y => y.DateTime.Month).OrderBy(u => u.DateTime.Year).OrderBy(r => r.CompanyName).OrderByDescending(t => t.CashTurnover);
```
[the remark this answers is not in the document]
Comment (sevap8) [reply HW5/Seva-Popov/Task/LinqSamples.cs]: Fix

File: HW5/Seva-Popov/Task/LinqSamples.cs (hunk 1 above)
Comment (sevap8) [reply HW5/Seva-Popov/Task/LinqSamples.cs]: Fix

File: HW5/Seva-Popov/Task/LinqSamples.cs (hunk 2 above)
Comment (sevap8) [reply HW5/Seva-Popov/Task/LinqSamples.cs]: Fix

File: HW5/Seva-Popov/Task/LinqToSqlSamples.cs (hunk 3 above)
Comment (sevap8) [reply HW5/Seva-Popov/Task/LinqToSqlSamples.cs]: Fix

File: HW5/Seva-Popov/Task/LinqToSqlSamples.cs (hunk 4 above)
Comment (sevap8) [reply HW5/Seva-Popov/Task/LinqToSqlSamples.cs]: ok

File: HW5/Seva-Popov/Task/LinqToSqlSamples.cs (hunk 5 above)
Comment (sevap8) [reply HW5/Seva-Popov/Task/LinqToSqlSamples.cs]: Fix, thanks for the comments

File: HW5/Seva-Popov/Task/LinqSamples.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (MilRoad) [reply HW5/Seva-Popov/Task/LinqSamples.cs]: I don't actually see the difference

File: HW5/Seva-Popov/Task/LinqSamples.cs [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+                ObjectDumper.Write(s);
+            }
+        }
+        
+
+        [Category("Homework")]
+        [Title("Homework - Task 10")]
+        [Description("Make the average annual statistics of customer activity by month (excluding the year), statistics by year, by year and month (i.e. when one month in different years has its own value).")]
+
+        public void Linq10()
```
Comment (MilRoad) [inline HW5/Seva-Popov/Task/LinqSamples.cs]: Why did you delete three others methods? They worked properly, you just needed to divide months by years count in the 10B method.
